src/pubtools/_impl/tracing.py

```diff
@@ -155,7 +193,26 @@ def force_flush(self):
             self._processor.force_flush()
         log.info("Flush trace data into OTEL collectors")
 
-    @property
-    def provider(self):
-        """Trace provider"""
-        return self._provider
+
+class CLITracingWrapper(TracingWrapper):
+    # Clients aren't expected to have open-telemetry. This version provides
+    # pass through functions to make open-telemetry a soft dependency.
+
+    # This class doesn't override reset, as it's intended for testing only.
+
+    def instrument_func(self, span_name=None, carrier=None, args_to_attr=False):
+        def _instrument_func(func):
+            @functools.wraps(func)
+            def wrap(_):
+                pass
+
+            return wrap
+
+        return _instrument_func
+
+    def force_flush(self):
+        """Flush trace data into OTEL collectors"""
+        log.warning(
```

Comment:
I don't think this warrants a warning, I think it can just be understood that `force_flush` does nothing if the feature is disabled.

The nature of the function is already that it can potentially do nothing (as there is not necessarily any data in need of flushing).